### PR TITLE
feat: vendir manager

### PR DESCRIPTION
**Pull request by dominik-bln**
## Changes

Adds a very basic vendir manager, initially only with the ability to update dependencies of the `helmChart` type.

## Context

- #17711

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/vendir/__fixtures__/empty-directories.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+apiVersion: vendir.k14s.io/v1alpha1
```
Comment (HonkingGoose) [inline lib/modules/manager/vendir/__fixtures__/empty-directories.yml:1]: Please create a `readme.md` file in this location: `lib/modules/manager/vendir/readme.md` and explain the limitations of the new Vendir manager, like this:

> The Vendir manager can only updates dependencies of the `helmChart` type.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/manager/vendir/readme.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The [vendir](https://carvel.dev/vendir/) manager only updates dependencies
+of the `helmChart` type.
```
Comment (HonkingGoose) [inline lib/modules/manager/vendir/readme.md:2]:
```suggestion
The [vendir](https://carvel.dev/vendir/) manager only updates dependencies of the `helmChart` type.
```

File: lib/modules/manager/vendir/extract.spec.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Fixtures } from '../../../../test/fixtures';
+import { extractPackageFile } from '.';
+
+const helmChart = Fixtures.get('helm-chart.yml');
+const emptyDirectories = Fixtures.get('empty-directories.yml');
```
Comment (rarkins) [inline lib/modules/manager/vendir/extract.spec.ts:5]: Please convert to inline codeBlock format

File: lib/modules/manager/vendir/extract.spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    });
+
+    it('returns null for empty directories key', () => {
+      const result = extractPackageFile(emptyDirectories);
+      expect(result).toBeNull();
+    });
+
+    it('extracts helm-chart from vendir.yml correctly', () => {
+      const result = extractPackageFile(helmChart);
+      expect(result).toMatchSnapshot({
```
Comment (rarkins) [inline lib/modules/manager/vendir/extract.spec.ts:30]:
```suggestion
      expect(result).toMatchObject({
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by rarkins** [comment]
Closed due to inactivity

**Pull request closed without merging**